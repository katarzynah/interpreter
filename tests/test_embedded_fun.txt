program f;

procedure x();
    var x,y :int;

    procedure z();
        var a,b : int;
        s: string;

        procedure hej();
            begin
                print("hej");
            end;

        procedure ho();
            begin
                print("ho");
            end;

        begin
            hej();
            ho();
            x := 15;
            a := 12;
        end;

    begin
        z();
        print(x);
        print(a);
    end;

begin
    x();
end.
