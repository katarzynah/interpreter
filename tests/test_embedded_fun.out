"ho"
"ho"
"ho"
"ho"
"ho"
"hej-ho"
"hej"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"ho"
"hej-ho"
"hej"
15
[(fromList [],fromList [(Ident "x",ProcDecProc (ProcHead (Ident "x") ArgsEmpty) (Dec (VarDecFull (VarDecListEnd (VarDecLabel (IdL (Ident "x") (IdLEnd (Ident "y"))) TypeSpecInt))) (ProcDecLabel (ProcDecProc (ProcHead (Ident "hej") ArgsEmpty) (Dec VarDecEmpty ProcDecEmpty) (CompStmnt (StmntList (SPrint (PrintStmnt (Act (ExpListOne (ExpSimple (SimpleExpTerm (TermFactor (FactorConstant (ConstString "hej"))))))))) StmntListEmpty))) (ProcDecLabel (ProcDecProc (ProcHead (Ident "z") ArgsEmpty) (Dec (VarDecFull (VarDecList (VarDecLabel (IdL (Ident "a") (IdLEnd (Ident "b"))) TypeSpecInt) (VarDecListEnd (VarDecLabel (IdLEnd (Ident "s")) TypeSpecString)))) (ProcDecLabel (ProcDecProc (ProcHead (Ident "ho") (Args (ArgListOne (ArgLabel (IdLEnd (Ident "c")) TypeSpecInt)))) (Dec VarDecEmpty ProcDecEmpty) (CompStmnt (StmntList (SIf (IfStmntWithElse (ExpSimple (SimpleExpTerm (TermFactor (FactorExpression (ExpGreater (SimpleExpTerm (TermFactor (FactorIdent (Ident "c")))) (SimpleExpTerm (TermFactor (FactorConstant (ConstInt 0))))))))) (SComp (CompStmnt (StmntList (SPrint (PrintStmnt (Act (ExpListOne (ExpSimple (SimpleExpTerm (TermFactor (FactorConstant (ConstString "ho"))))))))) (StmntList (SProc (ProcCall (Ident "ho") (Act (ExpListOne (ExpSimple (SimpleExpSubst (SimpleExpTerm (TermFactor (FactorIdent (Ident "c")))) (TermFactor (FactorConstant (ConstInt 1))))))))) StmntListEmpty)))) (SPrint (PrintStmnt (Act (ExpListOne (ExpSimple (SimpleExpTerm (TermFactor (FactorConstant (ConstString "hej-ho"))))))))))) StmntListEmpty))) ProcDecEmpty)) (CompStmnt (StmntList (SProc (ProcCall (Ident "ho") (Act (ExpListOne (ExpSimple (SimpleExpTerm (TermFactor (FactorIdent (Ident "x"))))))))) (StmntList (SAss (AssStmnt (Ident "x") (ExpSimple (SimpleExpTerm (TermFactor (FactorConstant (ConstInt 15))))))) (StmntList (SAss (AssStmnt (Ident "a") (ExpSimple (SimpleExpTerm (TermFactor (FactorConstant (ConstInt 12))))))) (StmntList (SProc (ProcCall (Ident "hej") ActEmpty)) StmntListEmpty)))))) ProcDecEmpty))) (CompStmnt (StmntList (SAss (AssStmnt (Ident "x") (ExpSimple (SimpleExpTerm (TermFactor (FactorConstant (ConstInt 5))))))) (StmntList (SProc (ProcCall (Ident "z") ActEmpty)) (StmntList (SProc (ProcCall (Ident "z") ActEmpty)) (StmntList (SPrint (PrintStmnt (Act (ExpListOne (ExpSimple (SimpleExpTerm (TermFactor (FactorIdent (Ident "x"))))))))) StmntListEmpty))))))])]
